Uncaught exception in nailgun handler

```
15:41:50:645 [INFO] notify invalidation: cleared 0 and dirtied 0 nodes for: {".pids/pantsd/fingerprint", ".pids/pantsd"}
15:41:50:714 [INFO] handling request: `--pants-config-files=+["/Users/asher/projects/toolchain/toolchain.toml"] --pants-bin-name=./tc test src/python/toolchain/buildsense/:: src/python/toolchain/django/site/::`
15:42:24:076 [INFO] request completed: `--pants-config-files=+["/Users/asher/projects/toolchain/toolchain.toml"] --pants-bin-name=./tc test src/python/toolchain/buildsense/:: src/python/toolchain/django/site/::`
15:42:24:079 [INFO] handling request: `--pants-config-files=+["/Users/asher/projects/toolchain/toolchain.toml"] --pants-bin-name=./tc fmt src/python/toolchain/buildsense/::`
15:42:33:225 [INFO] request completed: `--pants-config-files=+["/Users/asher/projects/toolchain/toolchain.toml"] --pants-bin-name=./tc fmt src/python/toolchain/buildsense/::`
15:42:33:229 [ERROR] Uncaught exception in nailgun handler: Throw {
    val: [Errno 9] Bad file descriptor,
    python_traceback: "Traceback (most recent call last):\n  File \"/Users/asher/.cache/pants/setup/bootstrap-Darwin-x86_64/1.29.0.dev1_py38/lib/python3.8/site-packages/pants/engine/internals/native.py\", line 288, in call\n    val = func(*args)\n  File \"/Users/asher/.cache/pants/setup/bootstrap-Darwin-x86_64/1.29.0.dev1_py38/lib/python3.8/site-packages/pants/bin/daemon_pants_runner.py\", line 173, in __call__\n    return self._run(working_directory.decode())\n  File \"/Users/asher/.pyenv/versions/3.8.2/lib/python3.8/contextlib.py\", line 120, in __exit__\n    next(self.gen)\n  File \"/Users/asher/.cache/pants/setup/bootstrap-Darwin-x86_64/1.29.0.dev1_py38/lib/python3.8/site-packages/pants/util/contextutil.py\", line 150, in stdio_as\n    yield\n  File \"/Users/asher/.pyenv/versions/3.8.2/lib/python3.8/contextlib.py\", line 120, in __exit__\n    next(self.gen)\n  File \"/Users/asher/.cache/pants/setup/bootstrap-Darwin-x86_64/1.29.0.dev1_py38/lib/python3.8/site-packages/pants/util/contextutil.py\", line 126, in _stdio_stream_as\n    new_dst.close()\nOSError: [Errno 9] Bad file descriptor\n",
    engine_traceback: [],
}


```

@stuhood

--- a/src/python/pants/util/contextutil.py
+++ b/src/python/pants/util/contextutil.py
@@ -121,9 +121,9 @@ def _stdio_stream_as(src_fd: int, dst_fd: int, dst_sys_attribute: str, mode: str
                 termios.tcdrain(dst_fd)
             else:
                 new_dst.flush()
+            new_dst.close()
         except BaseException:
             pass
-        new_dst.close()
 
         # Restore the python and os level file handles.
         os.dup2(old_dst_fd, dst_fd)

